server/vcs/provider/github/file_fetcher.go

```diff
@@ -52,3 +67,40 @@ func (r *RemoteFileFetcher) GetModifiedFilesFromCommit(ctx context.Context, repo
 	}
 	return files, nil
 }
+
+// GetModifiedFilesFromPR returns the names of files that were modified in the pull request
+// relative to the repo root, e.g. parent/child/file.txt.
+func (r *RemoteFileFetcher) getModifiedFilesFromPR(ctx context.Context, repo models.Repo, pullNum int, installationToken int64) ([]string, error) {
+	var files []string
+	nextPage := 0
+	for {
```

Comment:
can you dedup the pagination? You could probably create a struct/method that takes a func and handles the pagination for you.

e.g.
```
type CommitDiffPaginator struct {
   call func(..) []*gh.CommitFile
}

func (p CommitDiffPaginator) Call(options gh.ListOptions) []*gh.CommitFile {
...
}
```